src/librustc_mir/borrow_check/nll/invalidation.rs

```diff
@@ -553,7 +553,7 @@ impl<'cg, 'cx, 'tcx, 'gcx> InvalidationGenerator<'cg, 'cx, 'tcx, 'gcx> {
 
     /// Generate a new invalidates(L, B) fact
     fn generate_invalidates(&mut self, b: BorrowIndex, l: Location) {
-        let lidx = self.location_table.mid_index(l);
+        let lidx = self.location_table.start_index(l);
```

Comment:
Why did you change this from the mid point to the start point?

Reply:
Because that's what we use here:

https://github.com/rust-lang/rust/blob/31789a658bb6b6c78da1f2b99a5f169e4e8b983b/src/librustc_mir/borrow_check/mod.rs#L959

I'm changing this in the invalidation facts, and not here, because we use the incoming borrow in the current implementation.

Reply:
OK. I don't actually think it matters much either way, because all the loans that are live on entry are also live at the mid point; I think though that we usually generate facts to "take effect" on the midpoint. (Conceptually, the mid-point is where the statement "has its effects", so that's the point that is doing the invalidation, but it's also valid to say that if -- on entry to the statement -- there are any borrows that it will mess up, that's en error.) Seems fine.